processor/src/test/java/org/mapstruct/ap/test/nullvaluepropertymapping/NullValuePropertyMappingTest.java

```diff
@@ -0,0 +1,143 @@
+    HomeDTO.class,
+    UserDTO.class
+})
+public class NullValuePropertyMappingTest {
+
+    @Test
+    @WithClasses(CustomerMapper.class)
+    @IssueKey("1306")
+    public void testStrategyAppliedOnForgedMethod() {
+
+        Customer customer = new Customer();
+        customer.setAddress( null );
+
+        UserDTO userDTO = new UserDTO();
+        userDTO.setHomeDTO( new HomeDTO() );
+        userDTO.getHomeDTO().setAddressDTO( new AddressDTO() );
+        userDTO.getHomeDTO().getAddressDTO().setHouseNo( 5 );
+        userDTO.setDetails( Arrays.asList( "green hair" ) );
+
+        CustomerMapper.INSTANCE.mapCustomer( customer, userDTO );
+
+        assertThat( userDTO.getHomeDTO() ).isNotNull();
+        assertThat( userDTO.getHomeDTO().getAddressDTO() ).isNotNull();
+        assertThat( userDTO.getHomeDTO().getAddressDTO().getHouseNo() ).isEqualTo( 5 );
+        assertThat( userDTO.getDetails() ).isNotNull();
+        assertThat( userDTO.getDetails() ).containsExactly( "green hair" );
+    }
+
+    @Test
+    @WithClasses({ NvpmsConfig.class, CustomerNvpmsOnConfigMapper.class })
+    public void testHierarchyIgnoreOnConfig() {
+        testConfig( ( Customer s, CustomerDTO t ) -> CustomerNvpmsOnConfigMapper.INSTANCE.map( s, t ) );
+    }
+
+    @Test
+    @WithClasses(CustomerNvpmsOnMapperMapper.class)
+    public void testHierarchyIgnoreOnMapping() {
+        testConfig( ( Customer s, CustomerDTO t ) -> CustomerNvpmsOnMapperMapper.INSTANCE.map( s, t ) );
+    }
+
+    @Test
+    @WithClasses(CustomerNvpmsOnBeanMappingMethodMapper.class)
+    public void testHierarchyIgnoreOnBeanMappingMethod() {
+        testConfig( ( Customer s, CustomerDTO t ) -> CustomerNvpmsOnBeanMappingMethodMapper.INSTANCE.map( s, t ) );
+    }
+
+    @Test
+    @WithClasses(CustomerNvpmsPropertyMappingMapper.class)
+    public void testHierarchyIgnoreOnPropertyMappingMehtod() {
+        testConfig( ( Customer s, CustomerDTO t ) -> CustomerNvpmsPropertyMappingMapper.INSTANCE.map( s, t ) );
+    }
+
+    @Test
+    @WithClasses(CustomerDefaultMapper.class)
+    public void testStrategyDefaultAppliedOnForgedMethod() {
+
+        Customer customer = new Customer();
+        customer.setAddress( null );
+
+        UserDTO userDTO = new UserDTO();
+        userDTO.setHomeDTO( new HomeDTO() );
+        userDTO.getHomeDTO().setAddressDTO( new AddressDTO() );
+        userDTO.getHomeDTO().getAddressDTO().setHouseNo( 5 );
+        userDTO.setDetails( Arrays.asList( "green hair" ) );
+
+        CustomerDefaultMapper.INSTANCE.mapCustomer( customer, userDTO );
+
+        assertThat( userDTO.getHomeDTO() ).isNotNull();
+        assertThat( userDTO.getHomeDTO().getAddressDTO() ).isNotNull();
+        assertThat( userDTO.getHomeDTO().getAddressDTO().getHouseNo() ).isNull();
+        assertThat( userDTO.getDetails() ).isNotNull();
+        assertThat( userDTO.getDetails() ).isEmpty();
+    }
+
+    @Test
+    @Ignore // test gives different results for JDK and JDT
```

Comment:
Still need to investigate this

Reply:
ok.. we take it a s a separate issue.

Reply:
JDK reports on method, Eclipse (correctly) on `@Mapping`..

Reply:
Can you create an issue (if you haven't done already). It would be interesting to see what happens in plain javac with this